cmds/core/tail/tail.go

```diff
@@ -134,7 +134,10 @@ func readLastLinesBackwards(input readAtSeeker, writer io.Writer, numLines uint)
 			return err
 		}
 		// merge this block to what was read so far
-		readData = append(buf[:n], readData...)
```

Comment:
I don't understand the benefit of this change.

Reply:
> I don't understand the benefit of this change.

`readData = append(buf[:n], readData...)`  means  append data to buf if `len(readData) + n < = cap(buf)` , then `readData` and `buf` share memory under the hood. So, `n, err := input.ReadAt(buf, pos)` will overwrite the first n bytes if  n < cap(buf), resulting in data loss.

Reply:
@rminnich Are there any other questions? Obviously, the data is messed up because of the reuse of slices, so data cannot be appended to `buf`.

Reply:
can't you just write 
readData = append(append(make([]byte, n, n+len(readData), buf[:n]...), readData...)
so it's a zero-length buf, with the needed capacity, which you append buf[:n] to, then append readData to that?

Reply:
Or, even, maybe we move to go 1.23 and then you can do this
readData = slices.Concat(buf[:n], readData)
that function just feels like something we don't want for long.